pkg/controller/toolchainstatus/toolchainstatus_controller.go

```diff
@@ -176,6 +184,35 @@ func (r *ReconcileToolchainStatus) aggregateAndUpdateStatus(reqLogger logr.Logge
 
 	// if any components were not ready then set the overall status to not ready
 	if len(unreadyComponents) > 0 {
+		// If the current ToolchainStatus:
```

Comment:
I don't see any tests for this toolchain status controller notification change. Can we add some please?

Reply:
Test added